libs/image-editor/ImageEditor/src/main/res/navigation/nav_graph.xml

```diff
@@ -21,6 +19,10 @@
                 android:name="outputFileExtension"
                 app:argType="string"
                 app:nullable="true"/>
+            <argument
```

Comment:
Hmm interesting, do we really need to define the same argument twice? Or is the first definition input and the second definition output?

Reply:
I referred [source frag nav graph args](https://github.com/android/architecture-samples/blob/master/app/src/main/res/navigation/nav_graph.xml#L30-L35), [dest frag nav graph args](https://github.com/android/architecture-samples/blob/master/app/src/main/res/navigation/nav_graph.xml#L64-L71) and [args fetch in the dest frag](https://github.com/android/architecture-samples/blob/master/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.kt#L42) in the architecture blueprints sample. 

So yes, they seem to be input and output args. 

The only difference is args are defined outside actions in the source frag in the sample. We can do that if we have multiple actions for a fragment needing the same args.

Reply:
It seems we don't use the output argument (shouldReturnToPreviewScreen) though, right?

Reply:
Sorry if I misunderstood your comment. So the source frag args are passed in the `PreviewImageFragmentDirections` action: actionPreviewFragmentToCropFragment

```
PreviewImageFragmentDirections.actionPreviewFragmentToCropFragment(
                    inputFilePath,
                    outputFileExtension,
                    shouldReturnToPreviewScreen
                )
```

and they're received  by the `CropFragment` in `CropFragmentArgs`. We do use `shouldReturnToPreviewScreen` in the `CropFragment`.

Am I missing something?

Reply:
Ohh, I see. That's a very weird API that we need to declare the same thing twice :D. I haven't noticed this until now. Thanks for the clarification!